clients/js/genjs.go

```diff
@@ -569,8 +569,13 @@ var singleParamMethodDefinitionTemplateString = `/**{{if .Description}}
   {{.MethodName}}({{range $param := .Params}}{{$param.JSName}}, {{end}}options) {
   {{- else}}
   {{.MethodName}}({{range $param := .Params}}{{$param.JSName}}, {{end}}options, cb) {
-    return this._hystrixCommand.execute(this._{{.MethodName}}, arguments);
+    let callback = cb;
+    if (!cb && typeof options === "function") {
+      callback = options;
+    }
+    return applyCallback(this._hystrixCommand.execute(this._{{.MethodName}}, arguments), callback);
```

Comment:
The core of the fix for both issues listed in the PR description involves lifting the callback application up to the top most level